### PR TITLE
Refactor StudentLogic.java #6563

### LINKED ISSUE #6563: Refactor StudentLogic.java
some method comments(such as validateTeams((List<StudentAttributes>, String)) do not follow java coding standard

### PR DESCRIPTION
**Pull request by boxin-yang**
Fixes #6563

### CODE REVIEW

File: src/main/java/teammates/logic/core/StudentsLogic.java [hunk 1]
```diff
@@ -592,7 +600,8 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return enrollmentDetails;
     }
     
-    /* All empty lines or lines with only white spaces will be skipped.
+    /*
+     * All empty lines or lines with only white spaces will be skipped.
      * The invalidity info returned are in HTML format.
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: This one doesn't follow the Javadoc standard. Should begin with `/**`

File: src/main/java/teammates/logic/core/StudentsLogic.java [hunk 2]
```diff
@@ -382,6 +389,7 @@ public void validateSectionsAndTeams(List<StudentAttributes> studentList, String
     
     /**
      * Validates teams for any team name violations
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: It's a good practice to end the first sentence (usually Javadoc summary) with period.

File: src/main/java/teammates/logic/core/StudentsLogic.java [hunk 3]
```diff
@@ -198,7 +199,8 @@ public void updateStudentCascade(String originalEmail, StudentAttributes student
         StudentAttributes originalStudent = getStudentForEmail(student.course, originalEmail);
         updateStudentCascadeWithSubmissionAdjustmentScheduled(originalEmail, student, hasDocument);
         
-        /* finalEmail is the string to be used to represent a student's email.
+        /*
+         * finalEmail is the string to be used to represent a student's email.
          * This is because:
          *  - originalEmail cannot be used when student's email is being updated with a new valid email
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: The markdown style formatting will not work as you expect. You can try to use html tag to format the javadoc.
More [here](http://www.oracle.com/technetwork/articles/java/index-137868.html).

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 2 above)
Comment (boxin-yang) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: good suggestion

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 1 above)
Comment (boxin-yang) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: This is a block comment not a Doc comment. Block comments start with /* where as Doc comments start with /**

Comment (boxin-yang) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: http://www.oracle.com/technetwork/java/codeconventions-141999.html#385

Comment (boxin-yang) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: This is a block comment not a doc comment. Block comment starts with /** while doc comment starts with /*

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 3 above)
Comment (boxin-yang) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: This is a block comment for developers to read and I think the format here is for viewers' convenience:)

Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: Oops. My bad. :P

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 1 above)
Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: But it describe what the method does right?

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 2 above)
Comment (xpdavid) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: And check other place also in this file.

File: src/main/java/teammates/logic/core/StudentsLogic.java (hunk 1 above)
Comment (samsontmr) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: Since it's directly above the method, consider making it a proper comment? :p

Comment (samsontmr) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: Since it's directly above the method, consider making it a proper comment? :p

**Comment by wkurniawan07** [comment]
Likely it won't be me to review this, but just to give some pointers:
- Multi-line comments with `//` instead of `/* ... */` is (was?) a deliberate decision so that anyone can easily comment out an entire method by adding `/*` at the very beginning of the method.
- When a comment is present before the method definition, it should strictly be Javadocs. If some don't fit that profile (i.e. describing a "how" or "why" instead of "what"), they should be moved into the method definition.

**Comment by boxin-yang** [comment]
@wkurniawan07  Thx for the suggestion! I think I will close this PR for now as Prof @damithc  has suggested that refactoring should be done only after the code is well covered with tests.

**Pull request closed without merging**